apps/teams-test-app/src/components/privateApis/ChatAPIs.tsx

```diff
@@ -35,10 +35,33 @@ const OpenChat = (): React.ReactElement =>
     },
   });
 
-const OpenGroupChat = (): React.ReactElement =>
+const OpenChat = (): React.ReactElement =>
+  ApiWithTextInput<OpenSingleChatRequest>({
+    name: 'openChat2',
+    title: 'Open Chat',
+    onClick: {
+      validateInput: (input) => {
+        if (!input.user || !input.message) {
+          throw new Error('Both user and message are required on the input');
+        }
+      },
+      submit: async (input) => {
+        return await chat
+          .openChat(input)
+          .then(() => {
+            return 'openChat was called';
+          })
+          .catch(() => {
+            return 'openChat' + noHostSdkMsg;
```

Comment:
It's not really that there wasn't any response, it's that the promise was rejected which means something went wrong. Right?

Reply:
True. My understanding was that we use the noHostSdkMsg for both cases but I will see how other APIs are handled.

Reply:
What did you find when you looked at other APIs?

Reply:
To me it feels like there are three possible scenarios:
1. We made the call to the host, and didn't hear back yet
2. We heard back and it's a success (yay!)
3. We heard back and got some errors back (sad :().

It feel like having 1. and 3. result in the same thing printed on the screen makes the tests not 100% reliable, because we can't really say if the error made it back to the app itself.

Reply:
I agree, I think we need to differentiate between all 1, 2, and 3. From looking at and testing various APIs like `setConfig` and `getAuthToken` as well as the structure in the test app components like `ApiWithTextInput` I think I better understand how the rejections and manually throwing errors affect the test app result. Could you confirm my following understanding?

```
// I think this is a correct way to write a test app element
await <some function that returns a promise>
setResult('complete')
```
I think: 
* If the above line 1 succeeds -> line 2 setResult('complete') is called
* If line 1 fails -> ApiWithTextInput catches it and calls line 59 in `apps/teams-test-app/src/components/utils/ApiWithTextInput.tsx`, which is `setResult('Error: ' + err);`. Hence the error is displayed

```js
// and so something like below is a bug, since noHostSdkMsg would be displayed only when `chat.openGroupChat(input)` call succeeds?
const OpenGroupChat = (): React.ReactElement =>
  ApiWithTextInput<OpenGroupChatRequest>({
    name: 'openGroupChat',
    title: 'Open Group Chat',
    onClick: {
      validateInput: (input) => {
        if (!input.users) {
          throw new Error('users is required on the input');
        }
      },
      submit: async (input) => {
        await chat.openGroupChat(input);
        return 'chat.openChat()' + noHostSdkMsg;
      },
    },
  });
```

Reply:
And with the first "correct" example I'm thinking the noHostSdkMsg is first displayed by the ApiWithTextInput and stays that way unless we have a new result from the call, so we usually wouldn't need to manually show the noHostSdkMsg, maybe not ever?

Reply:
That matches my understanding.

Reply:
I thought I left a couple comments after this but I guess it didn't actually get posted 🤦🏼‍♀️  thank you for the knowledge, I basically had just said I agree all 1 2 3 should be covered with appropriate responses for each and listed a few examples I found to understand how to do so correctly. Since the PR was approved I'm assuming I did it correctly 😋